overlord/configstate/configcore/kernel.go

```diff
@@ -0,0 +1,165 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+//go:build !nomanagers
+// +build !nomanagers
+
+/*
+ * Copyright (C) 2023 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package configcore
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/snapcore/snapd/asserts"
+	"github.com/snapcore/snapd/i18n"
+	"github.com/snapcore/snapd/logger"
+	"github.com/snapcore/snapd/osutil"
+	"github.com/snapcore/snapd/overlord/configstate/config"
+	"github.com/snapcore/snapd/overlord/devicestate"
+	"github.com/snapcore/snapd/overlord/snapstate"
+	"github.com/snapcore/snapd/overlord/state"
+)
+
+const (
+	OptionKernelCmdlineAppend              = "system.kernel.cmdline-append"
+	OptionKernelDangerousCmdlineAppend     = "system.kernel.dangerous-cmdline-append"
+	CoreOptionKernelCmdlineAppend          = "core.system.kernel.cmdline-append"
+	CoreOptionKernelDangerousCmdlineAppend = "core.system.kernel.dangerous-cmdline-append"
+)
+
+func init() {
+	supportedConfigurations[CoreOptionKernelCmdlineAppend] = true
+	supportedConfigurations[CoreOptionKernelDangerousCmdlineAppend] = true
+}
+
+func changedKernelConfigs(c RunTransaction) []string {
+	changed := []string{}
+	for _, name := range c.Changes() {
+		// Note that we cannot just check the prefix as we have
+		// system.kernel.* options also defined in sysctl.go.
+		if name == CoreOptionKernelCmdlineAppend || name == CoreOptionKernelDangerousCmdlineAppend {
+			nameWithoutSnap := strings.SplitN(name, ".", 2)[1]
+			changed = append(changed, nameWithoutSnap)
+		}
+	}
+	return changed
+}
+
+func validateParamsAreAllowed(st *state.State, devCtx snapstate.DeviceContext, params []string) error {
+	// gd, err := devicestate.CurrentGadgetInfo(st, devCtx)
+	// if err != nil {
+	// 	return err
+	// }
+	// logger.Debugf("gadget data read from %s", gd.RootDir)
+	// TODO use gadgetdata to check against allowed values
+
+	return nil
+}
+
+func validateCmdlineExtra(c RunTransaction) error {
```

Comment:
we probably should rename things to match the new terminology as we don't use extra anymore